### PR TITLE
[Explicit Module Builds] Fix topological sort of inter-module dependency graph to account for Swift overlay dependencies

### PR DESCRIPTION
**Pull request by artemcm**

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci test

File: Sources/SwiftDriver/ExplicitModuleBuilds/InterModuleDependencies/InterModuleDependencyGraph.swift [hunk 1]
```diff
@@ -31,7 +31,7 @@ public enum ModuleDependencyId: Hashable {
     }
   }
 
-  internal var moduleNameForDiagnostic: String {
+  var moduleNameForDiagnostic: String {
```
Comment (cachemeifyoucan) [inline Sources/SwiftDriver/ExplicitModuleBuilds/InterModuleDependencies/InterModuleDependencyGraph.swift:34]: Is this used anywhere?

**Review by cachemeifyoucan — APPROVED** [review]
LGTM other than one small question.

**Comment by artemcm** [comment]
@swift-ci test

File: Sources/SwiftDriver/ExplicitModuleBuilds/InterModuleDependencies/InterModuleDependencyGraph.swift (hunk 1 above)
Comment (artemcm) [reply Sources/SwiftDriver/ExplicitModuleBuilds/InterModuleDependencies/InterModuleDependencyGraph.swift]: We use this in incremental build explanation remarks. Though the change here was not intentional so I reverted it.

**Comment by artemcm** [comment]
@swift-ci test Windows platform

**Pull request merged**